pyemu/utils/helpers.py

```diff
@@ -235,6 +235,137 @@ def autocorrelated_draw(pst,struct_dict,time_distance_col="distance",num_reals=1
             full_oe.loc[:, n] = v
     return full_oe
 
+def draw_by_group(pst, num_reals=100, sigma_range=6, use_specsim=False,
+         struct_dict={}, delr=[], delc=[], scale_offset=True, echo=True, logger=False):
+    """Draw a parameter ensemble from the distribution implied by the initial parameter values in the
+    control file and the prior parameter covariance matrix. Previously in pst_from.
+
+    Args:
+        pst ():
+        num_reals (`int`): the number of realizations to draw
+        sigma_range (`int`): number of standard deviations represented by parameter bounds.  Default is 6 (99%
+            confidence).  4 would be approximately 95% confidence bounds
+        use_specsim (`bool`): flag to use spectral simulation for grid-scale pars (highly recommended).
+            Default is False
+        struct_dict (`dict`): struct_dict (`dict`): a dict of GeoStruct (or structure file), and list of
+            pilot point template files pairs. If the values in the dict are
+            `pd.DataFrames`, then they must have an 'x','y', and 'parnme' column.
+            If the filename ends in '.csv', then a pd.DataFrame is loaded,
+            otherwise a pilot points file is loaded.
```

Comment:
Yes, this is critical. I copied the language straight from "geostatistical_draws" in an effort to be consistent and since that is where struct_dict gets passed. I probably missed something. I agree it could be made more flexible, but I wanted to minimize changes until I was sure the extraction was successful. Also worried about chasing any changes through the rest and breaking something!

Reply:
Ah yeah, I see that downstream greostruct_draws does accept tpl-csv pairs. unfortunately the specsim method in this method is expecting par_data_dfs so the tpl-pairs won't work if using specsim... This is another enhancement we could add down the line. For now might be better to have struct_dict here as a dict of lists of dfs (e.g. {GeoStruct: list[par_dfs], ...}) and not the optional use of til-input file pairs, if not using specsim...